projects/common/lib/components/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,76 @@
+import { forwardRef, Component, OnInit, Input, Output, EventEmitter, ViewChild, ElementRef} from '@angular/core';
+import { Base } from '../../../models/src/base';
+import { ControlContainer, ControlValueAccessor, NgForm, NG_VALUE_ACCESSOR } from '@angular/forms';
+/**
+ * Checkbox component is an input checkbox.  
+ * 
+ * @example
+ *       <common-checkbox #addressChangeChkBx 
+ *          [label]="'Do you want to opt in?'" 
+ *          (dataChange)="dataChange($event)" 
+ *          [model]="hasOpted" [disabled]="isDisabled" 
+ *          [required]="isrequired">
+ *       </common-checkbox>
+ *      
+ * @export
+ */
+
+
+@Component({
+  selector: 'common-checkbox',
+  templateUrl: './checkbox.component.html',
+  styleUrls: ['./checkbox.component.scss'],
+  viewProviders: [
+    { provide: ControlContainer, useExisting: forwardRef(() => NgForm ) }
+  ],
+  providers: [
+    { provide: NG_VALUE_ACCESSOR, multi: true, useExisting: forwardRef(() => CheckboxComponent )}
+  ]
+})
+
+export class CheckboxComponent extends Base implements  ControlValueAccessor {
+
+  @Input() model: boolean = true ;
+  @Input() required: boolean = false;
+  @Input() disabled: boolean = false;
+  @Input() label: string = 'Default Checkbox';
+  @Input() errorMessageRequired: string = this.label +'Field is required.';
+  @Input() checked: boolean =  false ;
+  @Input() showError: boolean;
+  @Output() dataChange: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
You know if you refactor this so you have an `@Input() data`, then because this one is called `dataChange` you get two-way data binding?  Look here - 

https://angular.io/guide/template-syntax#two-way-binding---

Essentially all you'd have to do is rename the `@Input() model` line to `@Input() data`, then because of the corresponding `dataChange` you could do:

<common-checkbox [(data)]='person.isWhatever'>

A lot of other components in the lib already do this, like MaskModel , ToggleComponent, and AddressComponent.